### PR TITLE
HAI-1603 Add a possibility to not actually send outgoing emails.

### PR DESCRIPTION
**Pull request by pitkni**
# Description

Add a possibility to not actually send outgoing emails.

application properties (and environment variables) can be used to control if filter is used and to what addresses emails will be sent to.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Other

# Instructions for testing
Please describe tests how this change or new feature can be tested.

# Checklist:

- [x] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location:

### CODE REVIEW

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/email/EmailSenderService.kt [hunk 1]
```diff
@@ -14,11 +14,14 @@ private val logger = KotlinLogging.logger {}
 @Service
 class EmailSenderService(
     private val mailSender: JavaMailSender,
-    @Value("\${haitaton.email.enabled}") private val enabled: Boolean,
+    @Value("\${haitaton.email.filter.use}") private val sendOnlyForAllowed: Boolean,
+    @Value("\${haitaton.email.filter.allow-list}") private val allowListRaw: String,
```
Comment (corvidian) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/email/EmailSenderService.kt:18]: You can use SpEL to split the property to a list directly: https://www.baeldung.com/spring-inject-arrays-lists#using-custom-delimiters

File: services/hanke-service/src/main/resources/application.properties [hunk 2]
```diff
@@ -86,6 +86,7 @@ spring.mail.properties.mail.debug=${MAIL_SENDER_DEBUG:false}
 spring.mail.properties.mail.smtp.auth=${MAIL_SENDER_AUTH:false}
 spring.mail.properties.mail.smtp.starttls.enable=${MAIL_SENDER_STARTTLS_ENABLE:true}
 
-haitaton.email.enabled=${HAITATON_EMAIL_ENABLED:false}
+haitaton.email.filter.use=${HAITATON_EMAIL_FILTER_USE:true}
```
Comment (corvidian) [inline services/hanke-service/src/main/resources/application.properties:89]: This should be set to false in docker-compose.yml, so we can still send email to all addresses in the docker environment.

Comment (pitkni) [reply services/hanke-service/src/main/resources/application.properties:90]: Fixed.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/email/EmailSenderService.kt (hunk 1 above)
Comment (pitkni) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/email/EmailSenderService.kt]: Nice, fixed.

**Review by corvidian — APPROVED** [review]

**Pull request merged**